qiskit_experiments/library/characterization/t2ramsey.py

```diff
@@ -117,6 +118,19 @@ def _set_backend(self, backend: Backend):
                 timing_constraints=timing_constraints, scheduling_method=scheduling_method
             )
 
+        # Set conversion factor
+        if self.experiment_options.unit == "dt":
+            try:
+                dt_factor = getattr(self.backend.configuration(), "dt")
+                conversion_factor = dt_factor
+            except AttributeError as no_dt:
+                raise AttributeError("Dt parameter is missing in backend configuration") from no_dt
+        elif self.experiment_options.unit != "s":
+            conversion_factor = apply_prefix(1, self.experiment_options.unit)
+        else:
+            conversion_factor = 1
+        self.set_experiment_options(conversion_factor=conversion_factor)
```

Comment:
It can be nice if there's a way to place this piece of code in a single location, instead of duplicating it in multiple experiments

Reply:
same for above